src/purifier/purifier.js

```diff
@@ -64,38 +54,157 @@ const PURIFY_PROFILES = /** @type {!DomPurifyConfig} */ ({
 });
 
 /**
- * Monotonically increasing counter used for keying nodes.
- * @private {number}
+ * @typedef {{addHook: !Function, removeAllHooks: !Function, sanitize: !Function}}
  */
-let KEY_COUNTER = 1;
+let DomPurifyDef;
 
 /**
- * Returns a <body> element containing the sanitized `dirty` markup.
- * Uses the standard DOMPurify config.
- * @param {string} dirty
- * @param {!Document} doc
- * @return {!Node}
+ * @typedef {function(string, string, string): string}
  */
-export function purifyHtml(dirty, doc) {
-  const config = standardPurifyConfig();
-  addPurifyHooks(DomPurify, doc);
-  const body = DomPurify.sanitize(dirty, config);
-  DomPurify.removeAllHooks();
-  return body;
-}
+export let AttributeRewrite;
```

Comment:
By convention, typedefs are suffixed with "Def". And since this is a function, maybe "rewriter"?

`AttributeRewriterDef`

Reply:
Good point, renamed.